app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala

```diff
@@ -290,6 +409,12 @@ class UndertakingController @Inject() (
         undertaking,
         timeProvider.now
       )
+      _ = {
+        undertakingJourney.addBusiness.value.getOrElse(List()).map { beEori =>
+          // ETMP has race condition where concurrent adds will overwrite eachother :shrug:
+          Await.result(escService.addMember(ref, BusinessEntity(beEori, false)), Duration(15, TimeUnit.SECONDS))
```

Comment:
Hmm....I don't think this is a good idea. Surely there has to be a better way. :/